metric_learn/constraints.py

```diff
@@ -33,6 +35,56 @@ def positive_negative_pairs(self, num_constraints, same_length=False,
       return a[:n], b[:n], c[:n], d[:n]
     return a, b, c, d
 
+  def generate_knntriplets(self, X, k_genuine, k_impostor):
+
+    labels = np.unique(self.partial_labels)
+    L = len(labels)
```

Comment:
avoid using cap symbol for something which is not a matrix. `n_labels` seems more appropriate

Reply:
Done